app/src/main/java/org/literacyapp/authentication/CameraViewActivity.java

```diff
@@ -182,23 +186,25 @@ public void onResume()
 
     private void storeStudentImages(){
         for(int i=0; i<studentImages.size(); i++){
-            String sId = collectionEventId + i;
-            MatName matName = new MatName(sId, studentImages.get(i));
+            String FileName = collectionEventId + String.format("%03d",i);
+            Long studentImageId = Long.parseLong(FileName);
+
+            MatName matName = new MatName(FileName, studentImages.get(i));
             FileHelper fh = new FileHelper();
             String wholeFolderPath = MultimediaHelper.getStudentImageDirectory() + "/" + deviceId + "/" + collectionEventId;
+
             new File(wholeFolderPath).mkdirs();
             fh.saveMatToImage(matName, wholeFolderPath + "/");
 
-            Long Id =  Long.parseLong(String.valueOf((int) (Math.random() * 1000000)));
-            StudentImage studentImage = new StudentImage(Id, null, wholeFolderPath, Calendar.getInstance(), null);
+            StudentImage studentImage = new StudentImage(studentImageId, null, wholeFolderPath + "/" + FileName + ".png", Calendar.getInstance(), null);
             studentImageDao.insert(studentImage);
         }
     }
 
     private void storeTestImages(){
         for(int i=0; i<testImages.size(); i++){
-            String sId = collectionEventId + i;
-            MatName matName = new MatName(sId, testImages.get(i));
+            String FileName = collectionEventId + String.format("%03d",i);
```

Comment:
@schaemik I just upgraded the greenDAO structure #204, so I had to touch `CameraViewActivity.java` because the `StudentImage` class was used there. So expect a possible conflict in that file when you update.